Sources/Contentful/ImageOptions.swift

```diff
@@ -347,3 +345,37 @@ private struct ImageParameters {
     static let quality          = "q"
     static let progressiveJPG   = "fl"
 }
+
+
+// Use CGColor instead of UIColor to enable cross-platform compatibility: macOS, iOS, tvOS, watchOS.
+internal extension CGColor {
+
+    // If for some reason the following code fails to create a hex string, the color black will be
+    // returned.
+    internal func hexRepresentation() -> String {
+        let hexForBlack = "000000"
+        guard let colorComponents = components else { return hexForBlack }
+        guard let colorSpace = colorSpace else { return hexForBlack }
+
+        let r, g, b: Float
+
+        switch colorSpace.model {
+        case .monochrome:
+            r = Float(colorComponents[0])
+            g = Float(colorComponents[0])
+            b = Float(colorComponents[0])
+
+        case .rgb:
+            r = Float(colorComponents[0])
+            g = Float(colorComponents[1])
+            b = Float(colorComponents[2])
+        default:
+            return hexForBlack
+        }
+
+        // Search the web for Swift UIColor to hex.
+        // This answer helped: https://stackoverflow.com/a/30967091/4068264
+        let hexString = String(format: "%02lX%02lX%02lX", lroundf(r * 255), lroundf(g * 255), lroundf(b * 255))
```

Comment:
Are you sure to use color components in the `[0-1]` range? If not, this would create strange outcomes, if for some reasons someone decides to use `r = 2`. 

Something like `min(1, max(0, x))` fore every color component `x` should do it.

Reply:
They are validated by the fact that the `CGColor` instance (i.e. `self`) exists.